scripts/tokenomics-emulation/ddc-smart-contract.ts

```diff
@@ -62,6 +67,82 @@ class DdcSmartContract {
     return txnObj;
   }
 
+  /**
+   * Deploy the smart contract to fetch code hash
+   * @param sender signer
+   * @returns Transaction hash
+   */
+  public async deploy(sender: KeyringPair) {
+    console.log(`Deploy DDC smart contract to get code hash`);
+    const code = new CodePromise(this.api, cere02Abi, cere02Wasm);
+
+    const tx = await code.createBlueprint();
+    return new Promise((res, rej) => {
+      tx.signAndSend(
+        sender,
+        Network.sendStatusCb.bind(this, res, rej)
+      ).catch((err) => rej(err));
+    });
+  }
+
+  /**
+   * Deploy the blluprint on chain
+   * @param sender smart contract owner
+   * @returns Transaction hash
+   */
+  public async bluePrint(
+    sender: KeyringPair,
+    codeHash: string,
+    endowment: string,
+    gasLimit: string,
+    tier1Limit: number,
+    tier1ThroughtputLimit: number,
+    tier1StorageLimit: number,
+    tier2Limit: number,
+    tier2ThroughtputLimit: number,
+    tier2StorageLimit: number,
+    tier3Limit: number,
+    tier3ThroughtputLimit: number,
+    tier3StorageLimit: number,
+    symbol: string
+  ) {
+    const blueprint = new BlueprintPromise(this.api, cere02Abi, codeHash);
+
+    const unsub = await blueprint.tx.new(
+      endowment,
+      gasLimit,
+      tier1Limit,
+      tier1ThroughtputLimit,
+      tier1StorageLimit,
+      tier2Limit,
+      tier2ThroughtputLimit,
+      tier2StorageLimit,
+      tier3Limit,
+      tier3ThroughtputLimit,
+      tier3StorageLimit,
+      symbol
+    );
+    return new Promise((res, rej) => {
+      unsub
+        .signAndSend(sender, Network.sendStatusCb.bind(this, res, rej, this.handleEvents))
+        .catch((err) => rej(err));
+    });
+  }
+
+  /**
+   * Process the events to set smart contract address
+   * @param events Events
+   */
+   private async handleEvents(events) {
+    console.log("Handling Events");
+    events.forEach((event) => {
+      if (event.event.data.length === 2) {
+        configFile.network.ddc_sc_address = event.event.data[1].toString();
```

Comment:
Will it be updated in config variable which already in use in the script?

Reply:
yes.